modules/launcher/src/test/java/org/mule/module/launcher/ResourceReleaserTestCase.java

```diff
@@ -91,6 +99,37 @@ public void deregisterJdbcDriversSameClassLoaders() throws Exception
         assertThat(list(getDrivers()), not(hasItem(jdbcDriver)));
     }
 
+    @Test
+    public void cleanUpResourcesBundleFromDisposedClassLoader() throws Exception
+    {
+        TestMuleApplicationClassLoader classLoader = new TestMuleApplicationClassLoader(new TestMuleSharedDomainClassLoader());
+        classLoader.setResourceReleaserClassLocation("/".concat(DefaultResourceReleaser.class.getName().replace(".", "/")).concat(".class"));
+
+        Map beforeCacheList = newHashMap();
+
+        Field cacheListField = ResourceBundle.class.getDeclaredField("cacheList");
+        cacheListField.setAccessible(true);
+        beforeCacheList.putAll((Map) cacheListField.get(null));
+
+        try
+        {
+            ResourceBundle.getBundle("aBundle", getDefault(), classLoader);
+        }
+        catch (MissingResourceException e)
+        {
+            // OK
+        }
+        Map actualCacheList = (Map) cacheListField.get(null);
+        int deltaForNonExistentBundle = actualCacheList.size() - beforeCacheList.size();
+
+        classLoader.dispose();
+
+        actualCacheList = (Map) cacheListField.get(null);
+
+        assertThat(actualCacheList.size(), equalTo(beforeCacheList.size() + deltaForNonExistentBundle));
+        assertThat(actualCacheList.entrySet(), (Matcher) hasItems(beforeCacheList.entrySet().toArray()));
```

Comment:
This question made me think about how I was testing this and how I implemented the clean up for the reference from the NONEXISTENT_BUNDLE, so I changed the implementation to instead of asking for a non existent bundle just use reflection to nullate the reference and now the test is readable and the state of the ResourceBundle after dispose doesn't have a "dirty" non-existent bundle. Please review this changes and let me know.